### PR TITLE
Return 201 when creation is successful

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

File: readinglist/resource.py [hunk 1]
```diff
@@ -176,6 +176,7 @@ def collection_post(self):
         new_record = self.request.validated
         new_record = self.process_record(new_record)
         self.record = self.db.create(record=new_record, **self.db_kwargs)
+        self.request.response.status = '201 Created'
```
Comment (tarekziade) [inline readinglist/resource.py]: I am not sure why we want to mess with status messages in our code base. This is to be dealt by the framework. We shoud set a 201 code and the thing should set the status automatically

Comment (tarekziade) [inline readinglist/resource.py]: see https://github.com/Pylons/webob/blob/master/webob/response.py#L274

```python
self.request.response.status_code = 201
```

File: readinglist/resource.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ def wrapped_view(self, *args, **kwargs):
                 # Transform the errors we got from colander into cornice errors
                 for field, error in e.asdict().items():
                     self.request.errors.add('body', field, error)
-            self.request.errors.status = "400 Bad Request"
+            self.request.errors.status = 400
```
Comment (ametaireau) [inline readinglist/resource.py]: why not `status_code` here?

File: readinglist/resource.py [hunk 3]
```diff
@@ -18,7 +18,7 @@ def wrapped_view(self, *args, **kwargs):
                 return view(self, *args, **kwargs)
             except RecordNotFoundError as e:
                 self.request.errors.add('path', 'id', six.text_type(e))
-                self.request.errors.status = "404 Resource Not Found"
+                self.request.errors.status = 404
```
Comment (ametaireau) [inline readinglist/resource.py]: why not `status_code` here?

Comment (Natim) [inline readinglist/resource.py]: Because it is request.errors and not response.status_code

Comment (Natim) [inline readinglist/resource.py]: This is the normal Cornice behavior: http://cornice.readthedocs.org/en/latest/validation.html#changing-the-status-code-from-validators

**Comment by Natim** [comment]
LGTM

**Pull request merged**